test/src/org/apache/jmeter/functions/TestTimeShiftFunction.java

```diff
@@ -0,0 +1,140 @@
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Collection;
+import java.util.TimeZone;
+
+import org.apache.jmeter.engine.util.CompoundVariable;
+import org.apache.jmeter.junit.JMeterTestCase;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.threads.JMeterContext;
+import org.apache.jmeter.threads.JMeterContextService;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestTimeShiftFunction extends JMeterTestCase {
+    private Function function;
+
+    private SampleResult result;
+
+    private JMeterVariables vars;
+
+    private JMeterContext jmctx = null;
+
+    private String value;
+
+    @Before
+    public void setUp() {
+        jmctx = JMeterContextService.getContext();
+        vars = new JMeterVariables();
+        jmctx.setVariables(vars);
+        jmctx.setPreviousResult(result);
+        result = new SampleResult();
+        function = new TimeShift();
+    }
+
+    @Test
+    public void testDatePlusOneDay() throws Exception {
+        Collection<CompoundVariable> params = makeParams("yyyy-dd-MM", "2017-01-01", "1d", "");
+        function.setParameters(params);
+        value = function.execute(result, null);
+        assertThat(value, is(equalTo("2017-02-01")));
+    }
+
+    @Test
+    public void testDatePlusOneDayInVariable() throws Exception {
+        Collection<CompoundVariable> params = makeParams("yyyy-dd-MM", "2017-01-01", "1d", "VAR");
+        function.setParameters(params);
+        function.execute(result, null);
+        assertThat(vars.get("VAR"), is(equalTo("2017-02-01")));
+    }
+
+    @Test
+    public void testDatePlusComplexPeriod() throws Exception {
+        Collection<CompoundVariable> params = makeParams("yyyy-dd-MM HH:m", "2017-01-01 12:00", "+1M+1d-1H-5m", "VAR");
+        function.setParameters(params);
+        String value = function.execute(result, null);
+        assertThat(value, is(equalTo("2017-02-02 10:55")));
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        Collection<CompoundVariable> params = makeParams("", "", "", "");
+        function.setParameters(params);
+        value = function.execute(result, null);
+        long resultat = Long.parseLong(value);
+        LocalDateTime  nowFromFunction = LocalDateTime.ofInstant(Instant.ofEpochMilli(resultat), TimeZone
+                .getDefault().toZoneId());
+        assertThat(nowFromFunction, within(5, ChronoUnit.SECONDS, LocalDateTime.now()));
+    }
+
+    @Test
+    public void testNowPlusOneDay() throws Exception {
+        Collection<CompoundVariable> params = makeParams("YYYY-MM-dd", "", "1d", "");
+        function.setParameters(params);
+        value = function.execute(result, null);
+        LocalDate tomorrow = LocalDate.now().plusDays(1);
+        LocalDate tomorrowFromFunction = LocalDate.parse(value);
```

Comment:
By default, parse use the format YYYY-MM-dd 
Javadoc : The string must represent a valid date and is parsed using DateTimeFormatter.ISO_LOCAL_DATE.